src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationSender.java

```diff
@@ -108,13 +115,20 @@ public void sendSms(
             client = notificationClient;
         }
 
-        SendSmsResponse sendSmsResponse = client.sendSms(
-                templateId,
-                phoneNumber,
-                personalisation,
-                reference,
-                smsSender
-        );
+        final SendSmsResponse sendSmsResponse;
+        try {
+            sendSmsResponse = client.sendSms(
+                    templateId,
+                    phoneNumber,
+                    personalisation,
+                    reference,
+                    smsSender
+            );
+        } catch (NotificationClientException e) {
```

Comment:
Can be combine to one catch ** (NotificationClientException | Exception e)** as it throw e and Exception will catch it

Reply:
Wanted to catch any exception apart from NotificationClientException. So that any other exception from gov notify can be wrapped in a NotificationClientException and be re-tried. 
We had recently faced a null pointer from gov notify when its response could not be determined. There is a bug outstanding in their code (https://github.com/alphagov/notifications-java-client/issues/171). We are catching this error so that it can be re-tried at a later time.

Reply:
to be clear this kind of code was **not** liked by sonar:
```java
try {
   ...
} catch (Exception e) {
    if (e instanceof NotificationClientException) {
      throw e;
   }
   throw new NotificationClientException(e);
}
```